src/include/sof/sof.h

```diff
@@ -48,6 +48,14 @@ struct sa;
 #define PP_NARG(...) (sizeof((unsigned int[]){0, ##__VA_ARGS__}) \
 	/ sizeof(unsigned int) - 1)
 
+/* compile-time assertion */
+#ifdef __XCC__
+	#define STATIC_ASSERT(COND, MESSAGE) \
+		typedef char assertion_failed_##MESSAGE[(COND) ? 1 : -1]
+#else
+	#define STATIC_ASSERT(COND, MESSAGE)
```

Comment:
Can we have a similar assertion for GCC i.e. emitting a #warning would probably work and be more portable ?

Reply:
I've dodged gcc emitting warning regarding unused-local-typedef by explicitly declaring it's attribute unused. This way we can have exact same assertion for both xcc and gcc.